src/core/value/constant.jl

```diff
@@ -79,11 +79,18 @@ Base.convert(::Type{T}, val::ConstantInt) where {T<:Signed} =
 end
 identify(::Type{Value}, ::Val{API.LLVMConstantFPValueKind}) = ConstantFP
 
-ConstantFP(typ::LLVMDouble, val::Real) =
+ConstantFP(typ::FloatingPointType, val::Real) =
     ConstantFP(API.LLVMConstReal(typ, Cdouble(val)))
 
-Base.convert(::Type{Float64}, val::ConstantFP) =
-    API.LLVMConstRealGetDouble(val, Ref{API.LLVMBool}())
+ConstantFP(val::Float16, ctx::Context=GlobalContext()) =
```

Comment:
we should probably not default to the `GlobalContext` and instead require an explicit context to be passed.

Reply:
I'm just following what `ConstantInt` does for uniformity; making `ctx` a required argument would be a breaking change.

Reply:
As Tim recently noted mixing `Context`s is a recipe for creating subtle issues.

Reply:
OK, pushed a change which should disable the default context.